include/message_buffer.h

```diff
@@ -143,7 +153,9 @@ typedef void * MessageBufferHandle_t;
  * \ingroup MessageBufferManagement
  */
 #define xMessageBufferCreate( xBufferSizeBytes ) \
-    ( MessageBufferHandle_t ) xStreamBufferGenericCreate( xBufferSizeBytes, ( size_t ) 0, pdTRUE )
+    ( MessageBufferHandle_t ) xStreamBufferGenericCreate( xBufferSizeBytes, ( size_t ) 0, pdTRUE, NULL, NULL )

```

Comment:
Wrapping `xStreamBufferGenericCreate` under `configUSE_SB_COMPLETED_CALLBACK` guard creates two function signatures dependent on if the configuration is enabled or not. This can lead to having two signatures for each of private functions.

> Either ways, xMessageBufferCreateWithCallback should be wrapped in configUSE_SB_COMPLETED_CALLBACK.

That makes sense. Wrapped the new APIs in configUSE_SB_COMPLETED_CALLBACK. This way user can know early as the compilation fails if they call `xMessageBufferCreateWithCallback` with configUSE_SB_COMPLETED_CALLBACK set to zero